src/tests/common/swc-transformer.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { readFileSync } = require('fs');
+const { join } = require('path');
+const { transformSync } = require('@swc/core');
+const { merge } = require('lodash');
+
+// This file can be replaced by @swc/jest once https://github.com/swc-project/jest/issues/8 resolves
+
+const swcrcPath = join(__dirname, '..', '..', '..', '.swcrc');
+const swcrc = JSON.parse(readFileSync(swcrcPath));
+
+const jestTransformOverrides = {
+    jsc: {
+        transform: {
+            hidden: {
+                jest: true,
+            },
+        },
+    },
+    module: {
+        type: 'commonjs',
+    },
+};
+
+const baseTransformOptions = merge({}, swcrc, jestTransformOverrides);
+
+module.exports = {
+    process: (src, path) => {
```

Comment:
It looks like Jest docs recommend implementing `[getCacheKey](https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object)` as well, although [swc-jest transformer](https://github.com/swc-project/jest/blob/master/index.ts) also doesn't seem to have it. Do you know if this would provide us any benefits?

Reply:
Excellent question; I didn't look at the jest transformer docs at all to be honest, I was mostly adapting from swc-jest's transformer (it would be better to contribute .swcrc support back to them, but I'm blocked on legal approval to sign their CLA before I can do that). I'll investigate this after standup!

Reply:
Discussed offline; will investigate improvements to the transformer in a separate PR (ideally against the swc-project/jest repo directly)